examples/platforms/cc2650/crypto/aes_alt.c

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2017, The OpenThread Authors.
+ * All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "mbedtls/aes.h"
+#include "aes_alt.h"
+
+#if defined(MBEDTLS_AES_ALT)
+
+#include <string.h>
+#include <driverlib/crypto.h>
+#include <driverlib/prcm.h>
+#include <common/code_utils.hpp>
+
+#define CC2650_AES_KEY_UNUSED (-1)
+#define CC2650_AES_CTX_MAGIC (0x7E)
+
+/**
+ * bitmap of which key stores are currently used
+ */
+static uint8_t used_keys = 0;
+
+/**
+ * number of active contexts, used for power on/off of the crypto core
+ */
+static unsigned int ref_num = 0;
+
+void mbedtls_aes_init(mbedtls_aes_context *ctx)
+{
+    if (ref_num++ == 0)
+    {
+        /* enable the crypto core */
+        /* The TRNG should already be running before we ever ask the AES core
+         * to do anything, if there is any scenario that the TRNG powers off
+         * the peripheral power domain use this code to repower it
+
+        PRCMPowerDomainOn(PRCM_DOMAIN_PERIPH);
+        while (PRCMPowerDomainStatus(PRCM_DOMAIN_PERIPH) != PRCM_DOMAIN_POWER_ON) {
+            ;
+        }
+        */
+        PRCMPeripheralRunEnable(PRCM_PERIPH_CRYPTO);
+        PRCMPeripheralSleepEnable(PRCM_PERIPH_CRYPTO);
+        PRCMPeripheralDeepSleepEnable(PRCM_PERIPH_CRYPTO);
+        PRCMLoadSet();
+
+        while (!PRCMLoadGet())
+        {
+            ;
+        }
```

Comment:
I'll change it to fit the coding convention. Is there a publicly available style sheet I can follow?